llama_stack/providers/inline/agents/meta_reference/agent_instance.py

```diff
@@ -510,16 +515,60 @@ async def _run(
             async with tracing.span("inference") as span:
                 if self.agent_config.name:
                     span.set_attribute("agent_name", self.agent_config.name)
-                async for chunk in await self.inference_api.chat_completion(
-                    self.agent_config.model,
-                    input_messages,
-                    tools=self.tool_defs,
-                    tool_prompt_format=self.agent_config.tool_config.tool_prompt_format,
-                    response_format=self.agent_config.response_format,
+                # Convert messages to OpenAI format
+                openai_messages = []
+                for message in input_messages:
+                    openai_message = await convert_message_to_openai_dict_new(message)
```

Comment:
I think this is what is causing the failure in the integration tests. When it gets down into the inference router and then the provider implementation, it's expecting a different type. 

the type here needs to be `OpenAIMessageParam` rather than a dict. I don't know if there is currently a helper for this, I kind of hit the same thing when I was doing some initial work in telemetry. You might need to augment `openai_compat.py` with a utility to go from Messages input -> [OpenAIMessageParam]

Reply:
Yes, thanks Charlie. That's what I was wondering as well.